eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/spanned/EmptyLinesTrimmed.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.utils.spanned;
+
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+
+/**
+ * {@link Spanned} decorator that trims leading and trailing empty lines.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class EmptyLinesTrimmed extends AbstractSpanned
+{
+    public EmptyLinesTrimmed(final Spanned input)
+    {
+        super(new Factory<Spanned>()
+        {
+            @Override
+            public Spanned create()
+            {
+                return trimSpannable(new SpannableStringBuilder(input));
+            }
+        });
+    }
+
+
+    // https://gist.github.com/mrmike/77fe2a6a357e78108115
+    private static SpannableStringBuilder trimSpannable(SpannableStringBuilder spannable)
+    {
+        int trimStart = 0;
+        int trimEnd = 0;
+
+        String text = spannable.toString();
+
+        while (text.length() > 0 && text.startsWith("\n"))
```

Comment:
Sorry, I missed the actual implementation before. It is rather inefficient as it creates a new String with every iteration. Consider using an implementation that works like in `Trimmed` (i.e. just count the number of chars that need to be removed instead of creating new strings without that char).